src/main/kotlin/xyz/atrius/waystones/data/config/Localization.kt

```diff
@@ -0,0 +1,42 @@
+package xyz.atrius.waystones.data.config
+
+import org.bukkit.configuration.file.YamlConfiguration
+import xyz.atrius.waystones.utility.KotlinPlugin
+import xyz.atrius.waystones.utility.translateColors
+import java.io.File
+import java.io.InputStreamReader
+import java.text.MessageFormat
+import java.util.*
+
+class Localization(val plugin: KotlinPlugin, localeTag: String) {
+    private var configFile = File(plugin.dataFolder, "locale-$localeTag.yml")
+    private val config = YamlConfiguration()
+    private var locale: Locale = Locale.getDefault()
+
+    init {
+        reload(localeTag)
+    }
+
+    fun localize(key: String, vararg args: Any?): String {
+        return MessageFormat(config.getString(key)!!, locale)
+                .format(*args.map { it ?: "null" }.toTypedArray()).translateColors()
```

Comment:
This feels somewhat inflexible to me. There is at least one case I can think of where we would want the value to be printed with a different value. Additionally, "null" wouldn't get localized into the appropriate languages.

Reply:
True. The infleibility can be somewhat mitigated tho, check the locale-en.yml. For `null`, maybe avoid passing null directly into the localization class? Instead ?: into a proper representation based on the context (e.g. unnamed-waystone: "Waystone").

Reply:
A fair point, I'd recommend going through things again to see if there is any place that might be occurring that could've been missed.